src/sync.cpp

```diff
@@ -179,15 +193,23 @@ void LeaveCritical()
 
 std::string LocksHeld()
 {
+    LockData& lockdata = GetLockData();
+    std::lock_guard<std::mutex> lock(lockdata.dd_mutex);
```

Comment:
Locking a non-recursive mutex that is already locked is undefined behavior

Reply:
[Fixed](https://github.com/bitcoin/bitcoin/pull/18881#issuecomment-630257992).